### PR TITLE
Implement read and write functionality for JSON

### PR DESCRIPTION
**Pull request by deepaksa1**
This PR is designed to introduce both reading and writing capabilities for JSON data type.
To test them, please execute `TestJsonRead` and `TestJsonWrite` under `JsonTest`

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs [hunk 1]
```diff
@@ -11478,10 +11484,18 @@ private Task WriteUnterminatedSqlValue(object value, MetaType type, int actualLe
...
                 case TdsEnums.SQLNTEXT:
                 case TdsEnums.SQLXMLTYPE:
+                case TdsEnums.SQLJSON:
 
                     if (type.IsPlp)
                     {
-                        if (IsBOMNeeded(type, value))
+                        if (type.NullableType == TdsEnums.SQLJSON)
+                        {
+                            // TODO : Performance and BOM check. Saurabh
```
Comment (apoorvdeshmukh) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs:11493]: I think we can remove this comment and instead mention the issue no as TODO which is tracking this improvement.  We can also consider addressing it in this PR if it's an easy fix.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                _output.WriteLine("Column DataType is " + column?.DataType.ToString());
+                _output.WriteLine("Column DataTypeName is " + column.DataTypeName);
+                Assert.Equal("json", column.DataTypeName);
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.IsJsonSupported))]
+        public void TestJsonWrite()
+        {       
+            string tableName = DataTestUtility.GetUniqueNameForSqlServer("Json_Test");
```
Comment (apoorvdeshmukh) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs:68]: You can consider calling DataTestUtility.DropTable() before creating table with your own name.

**Review by apoorvdeshmukh — COMMENTED** [review]
Suggested few minor changes.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs (hunk 1 above)
Comment (deepaksa1) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs]: Done.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs (hunk 2 above)
Comment (deepaksa1) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs:68]: `DataTestUtility.GetUniqueNameForSqlServer` ensures a unique name for the table. We can modify it to use `DataTestUtility.CreateTable()` once the streaming PR is checked into main [https://github.com/dotnet/SqlClient/pull/2801](url)

**Review by apoorvdeshmukh — APPROVED** [review]
As discussed, please create/update the issue tracking additional enhancements to the testcases.

File: doc/snippets/Microsoft.Data/SqlDbTypeExtensions.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## Remarks  
+The <xref:Microsoft.Data.SqlDbTypeExtensions> class provides <xref:System.Data.SqlDbType> enums which will be added. These are meant to be used by applications which cannot leverage the enums in <xref:System.Data.SqlDbType> available in newer .NET runtime versions.
+
+]]>
+        </format>
+      </remarks>
+    </SqlDbTypeExtensions>
+    <SqlJson name="default">
+      <summary>Gets the <see cref="T:System.Data.SqlDbType" /> enum value for JSON datatype.</summary>
```
Comment (cheenamalhotra) [inline doc/snippets/Microsoft.Data/SqlDbTypeExtensions.xml:18]:
```suggestion
      <summary>Gets the <see cref="T:System.Data.SqlDbType" /> enum value for the JSON datatype.</summary>
```

**Review by cheenamalhotra — APPROVED** [review]

**Pull request merged**